repos/system_upgrade/el7toel8/libraries/tcpwrappersutils.py

```diff
@@ -0,0 +1,80 @@
+import re
+
+
+def _build_regex(pattern):
+    regex = '^'
+    part_beginning = 0
+    while part_beginning < len(pattern):
+        ix1 = pattern.find('*', part_beginning)
+        ix2 = pattern.find('?', part_beginning)
+        ix1 = len(pattern) if ix1 < 0 else ix1
+        ix2 = len(pattern) if ix2 < 0 else ix2
+        part_end = min(ix1, ix2)
+
+        regex += re.escape(pattern[part_beginning:part_end])
+
+        if part_end < len(pattern):
+            if pattern[part_end] == '*':
```

Comment:
nit: Squash L17-L20 to ternary operator?
```regex += '.*' if pattern[part_end] == '*' else '.'```

Reply:
I find my version more readable.